app/src/main/kotlin/com/simplemobiletools/clock/services/TimerService.kt

```diff
@@ -34,19 +37,44 @@ class TimerService : Service() {
 
     override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
         super.onStartCommand(intent, flags, startId)
-
-        val formattedDuration = config.timerSeconds.getFormattedDuration()
-        startForeground(TIMER_RUNNING_NOTIF_ID, notification(formattedDuration))
-
+        isStopping = false
+        updateNotification()
+        startForeground(TIMER_RUNNING_NOTIF_ID, notification(getString(R.string.app_name), getString(R.string.timer_notification_msg), INVALID_TIMER_ID))
         return START_NOT_STICKY
     }
 
+    private fun updateNotification() {
+        timerHelper.getTimers { timers ->
+            val runningTimers = timers.filter { it.state is TimerState.Running }
+            if (runningTimers.isNotEmpty()) {
+                val firstTimer = runningTimers.first()
+                val formattedDuration = (firstTimer.state as TimerState.Running).tick.getFormattedDuration()
+                val contextText = when {
+                    runningTimers.size > 1 -> getString(R.string.timer_multiple_notification_msg, runningTimers.size)
+                    firstTimer.label.isNotEmpty() -> getString(R.string.timer_single_notification_label_msg, firstTimer.label)
+                    else -> getString(R.string.timer_single_notification_msg, runningTimers.size)
```

Comment:
The above timer_multiple_notification_msg has to be added as a Plural in the strings file so it can be properly translated in case it is 2 timers or 5, as it differs in some languages. I mean like https://github.com/SimpleMobileTools/Simple-Commons/blob/7c0a8d628ee23094e669a6169220b8da0fe9e649/commons/src/main/res/values/strings.xml#L165 . Add the "one" usecase too, then you can remove the timer_single_notification_msg string